resources/persist/src/lib.rs

```diff
@@ -41,6 +47,47 @@ impl PersistInstance {
         Ok(serialize_into(&mut writer, &struc).map_err(PersistError::Serialize))?
     }
 
+    /// list method returns a vector of strings containing all the keys associated with a PersistInstance
+    pub fn list(&self) -> Result<Vec<String>, PersistError> {
+        let storage_folder = self.get_storage_folder();
+
+        let mut list = Vec::new();
+
+        let entries = fs::read_dir(storage_folder).map_err(PersistError::ListFolder)?;
+        for entry in entries {
+            let key = entry.map_err(PersistError::ListFolder)?;
+            let key_name = key
+                .path()
+                .file_stem()
+                .unwrap_or_default()
+                .to_str()
+                .unwrap_or("file name contains non-UTF-8 characters")
```

Comment:
This one is causing a bit of head bashing, but will continue to bash. I'm not seeing how to propagate right in this moment, but will get there.

Reply:
You can take a shortcut and map the error to a `ListFolder` and propagate with `?` like the statement before it.

Reply:
Aha, well then don't you already have a `Result<&str, PersistError>` after doing `.ok_or(PersistError::ListFolder)`? If so, you can do the `?` immediately, then `to_string()`.

Reply:
Hmmmmm...might just have to take my laptop with me to work today and continue to work on this at lunch :)

Reply:
Ok, I think I've got this worked out. I figured out how to use ok_or to convert the error related to the potential for invalid Unicode characters.  Now I just have to work out a test...

Reply:
I would say that error does not need a test case, since it should realistically never happen. Since the keys (file names) are `String`s when they are saved, they should be valid utf-8 when reading the directory as well.